src/auth/verification-code/verification-code.service.ts

```diff
@@ -0,0 +1,69 @@
+import { ForbiddenException, Inject, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { VerificationCode } from './verification-code.entity';
+import { Repository } from 'typeorm';
+import { User } from 'src/user/user.entity';
+import * as moment from 'moment';
+import { generateVerificationCode } from '../utils/verification-code-generator';
+import {
+  VERIFICATION_CODE_MINUTES_VALIDITY,
+  VERIFICATION_EMAIL_SUBJECT,
+  VERIFICATION_EMAIL_TEXT,
+} from '../utils/config-constants';
+import { NodeMailerService } from 'src/node-mailer/node-mailer.service';
+
+@Injectable()
+export class VerificationCodeService {
+  constructor(
+    @InjectRepository(VerificationCode)
+    private verificationCodeRepository: Repository<VerificationCode>,
+    @Inject(NodeMailerService) private nodeMailerService: NodeMailerService,
+  ) {}
+
+  createOne(user: User) {
+    const verificationCode = new VerificationCode();
+    verificationCode.user = user;
+    verificationCode.code = generateVerificationCode();
+    verificationCode.validUntil = new Date(
+      moment().add(VERIFICATION_CODE_MINUTES_VALIDITY, 'minutes').toString(),
+    );
+    verificationCode.used = false;
+
+    return this.verificationCodeRepository.save(verificationCode);
+  }
+
+  sendOne(verificationCode: VerificationCode) {
+    return this.nodeMailerService.sendEmail(
+      verificationCode.user.email,
+      VERIFICATION_EMAIL_SUBJECT,
+      VERIFICATION_EMAIL_TEXT(verificationCode.code),
+    );
+  }
+
+  async createAndSendOne(user: User) {
+    const verificationCode = await this.createOne(user);
+
+    await this.sendOne(verificationCode);
```

Comment:
where do we handle failure here ?

Reply:
Let's use the this above [thread](https://github.com/hossamnasser938/ECommerce-BND/pull/2#discussion_r1299217325) to track this discussion